pkg/services/sqlstore/sqlstore.go

```diff
@@ -423,6 +431,31 @@ func (ss *SQLStore) initEngine(engine *xorm.Engine) error {
 	return nil
 }
 
+// The transaction_isolation system variable isn't compatible with MySQL < 5.7.20 or MariaDB. If we get an error saying this
+// system variable is unknown, then replace it with it's older version tx_isolation which is compatible with MySQL < 5.7.20 and MariaDB.
+func (ss *SQLStore) ensureTransactionIsolationCompatibility(engine *xorm.Engine, connectionString string) (*xorm.Engine, error) {
+	var result string
+	_, err := engine.SQL("SELECT 1").Get(&result)
+
+	var mysqlError *mysql.MySQLError
+	if errors.As(err, &mysqlError) {
+		// if there was an error due to transaction isolation
+		if strings.Contains(mysqlError.Message, "Unknown system variable 'transaction_isolation'") {
+			// replace with compatible system var for transaction isolation
```

Comment:
nit; it would be nice to add a debug message informing about the overriding of the connection string

Reply:
nice, yeah good idea. Will add